### PR TITLE
Integrates Shams's feedback on flows and APIs.

### PR DESCRIPTION
**Pull request by joeldudleyr3**

### CODE REVIEW

File: java-source/src/main/java/com/example/api/ExampleApi.java [hunk 1]
```diff
@@ -30,6 +30,8 @@
     private final String myLegalName;
     private final List<String> notaryNames = Lists.newArrayList("Controller", "NetworkMapService");
 
+    private final Logger logger = LoggerFactory.getLogger(ExampleApi.class);
```
Comment (shamsasari) [inline java-source/src/main/java/com/example/api/ExampleApi.java]: static

File: java-source/src/main/java/com/example/api/ExampleApi.java [hunk 2]
```diff
@@ -108,14 +110,8 @@ public Response createIOU(IOU iou, @PathParam("party") String partyName) throws
...
-
-            if (ex.getCause() instanceof FlowSessionException) {
-                msg = "Counterparty flow terminated unexpectedly.";
-            } else if (ex.getCause() instanceof RuntimeException) {
-                msg = ex.getCause().getMessage();
-            } else {
-                msg = "Unexpected error.";
-            }
+            msg = "Transaction failed.";
+            logger.error(ex.getMessage());
```
Comment (shamsasari) [inline java-source/src/main/java/com/example/api/ExampleApi.java]: `logger.error("Unable to create IOU", e)` - key bit is the second param so we get the stack trace in our logs

File: java-source/src/main/java/com/example/flow/ExampleFlow.java [hunk 3]
```diff
@@ -174,8 +170,8 @@ public SignedTransaction call() throws FlowException {
                             // We want to be sure that the agreed-upon IOU is valid under the rules of the contract.
                             // To do this we need to run the contract's verify() function.
                             wireTx.toLedgerTransaction(getServiceHub()).verify();
-                        } catch (SignatureException|TransactionResolutionException ex) {
-                            throw new RuntimeException(ex);
+                        } catch (SignatureException ex) {
+                            throw new FlowException(ex);
```
Comment (shamsasari) [inline java-source/src/main/java/com/example/flow/ExampleFlow.java]: `throw new FlowException(tx.getId() + " failed signature checks", ex);`

File: java-source/src/test/java/com/example/contract/IOUFlowTests.java [hunk 4]
```diff
@@ -52,8 +52,7 @@ public void flowReturnsASignedTransaction() throws InterruptedException, Executi
         ListenableFuture<SignedTransaction> future = a.getServices().startFlow(flow).getResultFuture();
         net.runNetwork(-1);
 
-        Object signedTx = future.get();
-        assert(signedTx instanceof SignedTransaction);
+        SignedTransaction signedTx = future.get();
```
Comment (shamsasari) [inline java-source/src/test/java/com/example/contract/IOUFlowTests.java]: Don't need to assign it to a variable that's not used. Compiler will warn about it.

File: kotlin-source/src/main/kotlin/com/example/api/ExampleApi.kt [hunk 5]
```diff
@@ -21,7 +22,9 @@ val NOTARY_NAMES = listOf("Controller", "NetworkMapService")
 // This API is accessible from /api/example. All paths specified below are relative to it.
 @Path("example")
 class ExampleApi(val services: CordaRPCOps) {
-    val myLegalName: String = services.nodeIdentity().legalIdentity.name
+    private val myLegalName: String = services.nodeIdentity().legalIdentity.name
+
+    private val logger = LoggerFactory.getLogger(ExampleApi::class.java)
```
Comment (shamsasari) [inline kotlin-source/src/main/kotlin/com/example/api/ExampleApi.kt]: inside a companion object and use `loggerFor`

File: kotlin-source/src/main/kotlin/com/example/api/ExampleApi.kt [hunk 6]
```diff
@@ -85,13 +88,8 @@ class ExampleApi(val services: CordaRPCOps) {
...
 
         } catch (ex: Throwable) {
-            val msg = when (ex) {
-                is ExecutionException -> ex.message
-                is FlowException -> ex.message
-                is FlowSessionException -> "Counterparty flow terminated unexpectedly."
-                else -> "Unexpected error."
-            }
-            Response.Status.BAD_REQUEST to msg
+            logger.error(ex.message)
```
Comment (shamsasari) [inline kotlin-source/src/main/kotlin/com/example/api/ExampleApi.kt]: See Java comment

File: java-source/src/test/java/com/example/contract/IOUFlowTests.java (hunk 4 above)
Comment (joeldudleyr3) [inline java-source/src/test/java/com/example/contract/IOUFlowTests.java]: Screw it, I'll drop the test. The compiler will force people to return a SignedTransaction anyway.

**Pull request merged**